src/api/c++/z3++.h

```diff
@@ -414,7 +454,16 @@ namespace z3 {
         context * m_ctx;
     public:
         object(context & c):m_ctx(&c) {}
-        object(object const & s):m_ctx(s.m_ctx) {}
+        object(object const &) = default;
+        object(object && s) noexcept : m_ctx{s.m_ctx} {
```

Comment:
this change and the assignment operator are useless.

Reply:
The noexcept is functional, but the `= default` is mostly just a coding practice to allow easier extension / editing in the future; they shouldn't really lead to code changes.

Reply:
See rule of zero here: https://en.cppreference.com/w/cpp/language/rule_of_three
More code is not better code.

Reply:
Ah, indeed there is no destructor, I can remove them all then since there is no need to set `m_ctx` to null then. Good catch.